### PR TITLE
Only convert omegaconf DictConfig to primitive type for logging

### PR DESCRIPTION
**Pull request by merelcht**
Signed-off-by: Merel Theisen <merel.theisen@quantumblack.com>

## Description
After merging https://github.com/kedro-org/kedro/pull/2174 where I made `OmegaConfLoader` return the primitive dict type I realised that in order to leverage other `omegaconf` functionality such as variable interpolation (https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#variable-interpolation) it's necessary to have it return the `omegaconf.dictconfig.DictConfig` type. 

## Development notes
The problems I discovered with logging were related to this line 
[`LOGGING.configure(logging_config)`](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/__init__.py#L258)

and so I decided to revert my previous change and only convert the `omegaconf.dictconfig.DictConfig` type to the primitive dict for the logging files. 

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2176"></a>

### CODE REVIEW

File: kedro/framework/session/session.py [hunk 1]
```diff
@@ -192,6 +193,8 @@ def create(  # pylint: disable=too-many-arguments
 
     def _get_logging_config(self) -> Dict[str, Any]:
         logging_config = self._get_config_loader()["logging"]
+        if isinstance(logging_config, omegaconf.DictConfig):
+            logging_config = OmegaConf.to_container(logging_config)
```
Comment (SajidAlamQB) [inline kedro/framework/session/session.py:197]: Is there a reason why logging config only works with it being a primitive container?

**Review by SajidAlamQB — APPROVED** [review]
Nice find 🕵️‍♀️, it all LGTM just had one question about the reason why this affected only the logging config.

File: kedro/framework/session/session.py (hunk 1 above)
Comment (merelcht) [reply kedro/framework/session/session.py:197]: Yes actually two reasons:
1. https://github.com/kedro-org/kedro/blob/main/kedro/framework/session/session.py#L197-L199 Needs a dict to properly convert the absolute paths
2. this line expects a regular dict: https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/__init__.py#L226

**Review by jmholzer — APPROVED** [review]
I appreciate the test in `test_session.py`. Looks like a clean solution.

File: kedro/config/omegaconf_config.py [hunk 2]
```diff
@@ -233,9 +233,8 @@ def load_and_merge_dir_config(self, conf_path: str, patterns: Iterable[str]):
 
         if aggregate_config:
             if len(aggregate_config) > 1:
-                merged_config = OmegaConf.merge(*aggregate_config)
-                return OmegaConf.to_container(merged_config)
-            return OmegaConf.to_container(list(aggregate_config)[0])
+                return dict(OmegaConf.merge(*aggregate_config))
+            return list(aggregate_config)[0]
         return {}
```
Comment (idanov) [inline kedro/config/omegaconf_config.py:238]: Could we flatten this up a bit? Nested ifs are harder to read, so maybe we can do something like:
```
if not aggregate_config:
    return {}
if len(aggregate_config)==1:
    return list(aggregate_config)[0]
return dict(OmegaConf.merge(*aggregate_config))
```

Comment (merelcht) [reply kedro/config/omegaconf_config.py]: Done!

**Pull request merged**